src/server/views/html/error.rs

```diff
@@ -20,3 +22,39 @@ pub fn render(title: &str, descr: &str) -> Response<Body> {
         },
     )
 }
+
+pub fn render_404() -> Response<Body> {
+    let rendered = html! {
+        html {
+            head {
+                meta charset="utf-8";
+                meta name="viewport" content="width=device-width, initial-scale=1";
+                title { "404 - Deps.rs" }
+                link rel="icon" type="image/svg+xml" href="/static/logo.svg";
+                link rel="stylesheet" type="text/css" href=(STATIC_STYLE_CSS_PATH);
+                link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Fira+Sans:400,500,600";
+                link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Source+Code+Pro";
+            }
+            body {
+                section class="hero is-light" {
+                    div class="hero-head" { (super::render_navbar()) }
+                }
+                section class="section" {
+                    div class="container" {
+                        div class="notification is-info" {
+                            p class="title is-3" { "Ooops, seems like you've hit a dead end!" }
+                            p { "The page you were looking for could not be found. In other words, this is a " b { "404 error" } "." }
+                        }
+                    }
+                }
+                (super::render_footer(None))
+            }
+        }
+    };
+
+    Response::builder()
+        .status(StatusCode::NOT_FOUND)
+        .header(CONTENT_TYPE, "text/html; charset=utf-8")
+        .body(Body::from(rendered.0))
```

Comment:
Unless anyone's opposed with this, it should be safe to tell the client to cache this page. To the best of my knowledge, rendered views are not cached by the server.
```suggestion
        .header(CACHE_CONTROL, "public, max-age=365000000, immutable")
        .body(Body::from(rendered.0))
```

Reply:
This seems far too long. I would limit it to only a few minutes.

Reply:
I set the caching time to 5 minutes which seems like a sane default.